### ISSUE TITLE
Kind support( pf-ring, ebpf, serviceMesh)

### ISSUE THREAD

**Issue by gsantoro** [kubeshark/kubeshark#1493]
**Describe the bug**
Kubeshark is not able to capture any traffic.

I get an error `ErrImagePull` from the daemon-set `kubeshark-worker-daemon-set` since the container image `kubeshark/pf-ring-module:all` hasn't been built for the architecture `linux/arm64/v8`. Because of this error, the worker is not initialized and no traffic is captured.

I can see other images have been built for my architecture but not this one unfortunately. 

Also Kind doesn't allow yet to load container images for other architectures. A fix for Kind has been already merged from the PR https://github.com/kubernetes-sigs/kind/pull/2957 but it hasn't been released yet.

**Provide more information**
Running on Kind on my Mac M1 laptop with architecture linux/arm64.

**To Reproduce**
1. Run `kubeshark tap`
2. Open the dashboard
4. See no traffic

**Expected behavior**
I expect to see some traffic being captured

**Logs**
Upload logs:
1. Run the kubeshark command with `--set dump-logs=true` (e.g `kubeshark tap --set dump-logs=true`)
2. Try to reproduce the issue
3. <kbd>CTRL</kbd>+<kbd>C</kbd> on terminal tab which runs `kubeshark`
4. Upload the logs zip file from `~/.kubeshark/kubeshark_logs_**.zip`

the previous command doesn't generate any logs for me

**Screenshots**
<img width="1420" alt="Screenshot 2024-02-01 at 16 22 02" src="https://github.com/kubeshark/kubeshark/assets/2378291/ef6f3592-ea2c-4331-a935-58f6a20fcb71">

**Desktop (please complete the following information):**
 - OS: macOS M1 with arm64 architecture
 - Web Browser: Google Chrome

**Additional context**

**Comment by alongir** [comment]
@gsantoro - while we look at this, can you run with `--set tap.kernelModule.enabled=false` ?

**Comment by gsantoro** [comment]
hey @alongir ,
I just tried that command:
- I can see the traffic being captured correctly
- the kernel module container is not present at all
- There is a CrashLoopBackOff from the tracer container with the following error logs

```
tracer/tls_process_discoverer.go:69 > Couldn't get the cgroup of process. error="Cgroup path not found for 387
```

**Comment by alongir** [comment]
@gsantoro can you let us know the kernel version, distribution, K8s version etc .. the cgroup issue sounds familiar (https://github.com/kubeshark/tracer/issues/20: fix coming up soon), but I'd like to make sure.

**Comment by alongir** [comment]
OK. It seems it's been a while since we tested Kubeshark on Minikube :-/
Thanks for bringing this to our attention!

The following command will operate on Mac/Minikube:
`kubeshark tap --set tap.kernelModule.enabled=false --set tap.tls=false --set tap.serviceMesh=false` - I validated this.

We'll also prioritize supporting Mac/Minikube.

## TL;DR

Both `ebpf`/`tls` and `serviceMesh` capabilities use the `CHECKPOINT_RESTORE` capability: https://github.com/kubeshark/kubeshark/blob/2b552b58470fc56d8df5983a423161a75bb9fcdd/helm-chart/values.yaml#L100.

Minikube on Mac doesn't recognize this capability.

**Comment by corest** [comment]
Updates regarding PF_RING

1. Recent kind VMs use kernel version not supported by PF_RING yet, so it is not possible to have it in kind as for now.
2. I've added multi-arch builds for pf-ring-module container. Currently `canary` tag version can be used on arm as well.

Tested on m1 mac with
```
kubeshark tap --set tap.kernelModule.image=kubeshark/pf-ring-module:canary
``` 

With this `load-pf-ring` init container will start without errors. It won't load any module because of 1, but that's expected.
I'll build this change into default `all` tag this week after few more tests

FYI @corest @gsantoro
